### ISSUE TITLE
[Feature Request] Export prop type interfaces such as `IButtonProps`

### ISSUE THREAD

**Issue by wenfangdu** [element-plus/element-plus#2040]
### Existing Component

Yes

### Component Name

All components

### Description

I'm encapsulating `ElButton` to `VButton`, instead of manually adding prop types, it'll be much more convenient to use the original `IButtonProps` instead, however, it's not exported:

```tsx
import { ElButton } from 'element-plus'
import { defineComponent, ref } from 'vue'

export default defineComponent({
  inheritAttrs: false,
  //           👇 manually adding prop types
  setup(props: { type: string; size: string }, { attrs, slots }) {
    // 👇 it'll be much more convenient to use `IButtonProps` instead
    // setup(props: IButtonProps, { attrs, slots }) {
    const btn = ref<InstanceType<typeof ElButton>>()

    return () => (
      <ElButton
        class='v-button'
        ref={btn}
        size='medium'
        onClick={() => btn.value?.$el.blur()}
        {...attrs}
        vSlots={slots}
      />
    )
  },
})
```

This request applies to all available components.

I'm relatively new to typescript, please correct me if it's infeasible. Also, if there's a better way for achieving the same typing experience without doing the exports, please let me know.

**Comment by JeremyWuuuuu** [comment]
Yeah,  I agree with you, this is something that we really need, but we are kind of in the situation of lack of people who can implement this. If you are willing to do this, that'd be great, but if you don't we will queue this task into our backlog in order that we can solve it when we got time.

**Comment by wenfangdu** [comment]
@JeremyWuuuuu Please queue it first.

**Comment by wellfrog16** [comment]
> Yeah, I agree with you, this is something that we really need, but we are kind of in the situation of lack of people who can implement this. If you are willing to do this, that'd be great, but if you don't we will queue this task into our backlog in order that we can solve it when we got time.

我也碰到了同样的问题。
最近在做vue2的迁移，之前vue2的一些组件大量使用mixin和解构props来复用element-ui的method、event和props
单纯的提供props的interface还不够
可能需要每个组件再提供一个hook用来复用method和event等
或许是一个很大的迭代

**Comment by sxzz** [comment]
We are working on it. There will be news to notify you.

**Comment by sxzz** [comment]
We have exported the props and types of some components.

```ts
import { buttonProps } from 'element-plus`
```
